R/yearmonth.R

```diff
@@ -65,8 +65,27 @@ yearmonth.Date <- function(x) {
 
 #' @export
 yearmonth.character <- function(x) {
-  assertDate(x)
-  new_yearmonth(anydate(x))
+  key_words <- regmatches(x, gregexpr("[[:alpha:]]+", x))
+  if (all(grepl("^[[:digit:]]{1~4}[[:space:]]*(m|mon|month)[[:space:]]*[[:digit:]]{1~4}$",
+                key_words, ignore.case = TRUE))) {
+    yr_mon <- regmatches(x, gregexpr("[[:digit:]]+", x))
+    digits_lgl <- map_lgl(yr_mon, ~ !has_length(.x, 2))
+    digits_len <- map_int(yr_mon, ~ sum(nchar(.x)))
+    digits_ind <- nchar(flatten_chr(yr_mon))
```

Comment:
What does this line exactly do? Why it can't be length of 3?

Reply:
If we allow years before 1000AD using this format, there is no way to tell the difference between %y and %Y for 2-digit years. Also, it seems more likely that a 3-digit year specification is an error than that it refers to a year before 1000AD (especially using this format).

Reply:
I think it's okay to throw an error for 3-digit years for now. If there's a compelling use case using 3-digit years, we'll accomodate this later.